### DISCUSSION TITLE
HuggingFaceInference API as LLM for dspy

### DISCUSSION THREAD

**Discussion by 97k** [stanfordnlp/dspy#738] [Q&A]
I loved the intro dspy notebook and helped me in understanding many underlying concepts fastly! I have been using HuggingFaceEndpoint in langchain that lets me call HuggingFaceInference API with repo_id and hugging_face_api_token, for all the LLMs that are available on their infrastructure for inference.

I know there is [HFClient TG](https://github.com/huggingface/text-generation-inference) but that creates a local server and I don't have that much resources to exhaust.

**Is it possible** to talk with cloud deployed [HF Inference API](https://huggingface.co/docs/api-inference/en/index) ?

**Comment by Josephrp** [comment]
i'm not a developper for DSPy or a contributor , yet, however this is within the scope of this issue , to a certain extent : #524 and if you're excited about it we can try to make these , starting with inference endpoint ... by the way the way DSPy currently works means we'll probably get very good results with the huggingface endpoints.

**Reply by 97k** [reply]
This is exciting!! Thanks for referring the PR.
I will walk through all the data points
